src/test/java/seedu/mtracker/filemanager/InstrumentDecoderTest.java

```diff
@@ -0,0 +1,154 @@
+    public static final String FUTURE_INVALID_DATE_STRING = "2 November";
+    public static final String TEST_REMARK = "";
+    public static final Instrument TEST_FOREX = new Forex(TEST_NAME, TEST_PRICE, TEST_SENTIMENT,
+            TEST_ENTRY_PRICE, TEST_EXIT_PRICE, FUTURE_DATE, TEST_REMARK);
+
+    public static final int NUMBER_OF_PARAMS = 8;
+    public static final int LENGTH_OF_DATE = 10;
+    public static final int LENGTH_OF_PRICES = 12;
+    public static final int LENGTH_OF_DONE = 5;
+    public static final int LENGTH_OF_INSTRUMENT = 5;
+
+    public static final String[] INVALID_INSTRUMENT_TYPE_TEXT_SEGMENT = {TEST_INVALID_TYPE,
+        TEST_NAME,
+        TEST_PRICE_STRING,
+        TEST_SENTIMENT,
+        TEST_DONE_STRING,
+        TEST_ENTRY_PRICE_STRING,
+        TEST_EXIT_PRICE_STRING,
+        FUTURE_DATE_STRING,
+        TEST_REMARK
+    };
+
+    public static final String[] INVALID_FOREX_NAME_TEXT_SEGMENT = {TEST_TYPE,
+        TEST_INVALID_FOREX_NAME,
+        TEST_PRICE_STRING,
+        TEST_SENTIMENT,
+        TEST_DONE_STRING,
+        TEST_ENTRY_PRICE_STRING,
+        TEST_EXIT_PRICE_STRING,
+        FUTURE_DATE_STRING,
+        TEST_REMARK
+    };
+
+    public static final String[] INVALID_FOREX_PRICE_TEXT_SEGMENT = {TEST_TYPE,
+        TEST_NAME,
+        TEST_INVALID_PRICE_STRING,
+        TEST_SENTIMENT,
+        TEST_DONE_STRING,
+        TEST_ENTRY_PRICE_STRING,
+        TEST_EXIT_PRICE_STRING,
+        FUTURE_DATE_STRING,
+        TEST_REMARK
+    };
+
+    public static final String[] INVALID_FOREX_SENTIMENT_TEXT_SEGMENT = {TEST_TYPE,
+        TEST_NAME,
+        TEST_PRICE_STRING,
+        TEST_INVALID_SENTIMENT,
+        TEST_DONE_STRING,
+        TEST_ENTRY_PRICE_STRING,
+        TEST_EXIT_PRICE_STRING,
+        FUTURE_DATE_STRING,
+        TEST_REMARK
+    };
+
+    public static final String[] INVALID_FOREX_EXPIRY_TEXT_SEGMENT = {TEST_TYPE,
+        TEST_NAME,
+        TEST_PRICE_STRING,
+        TEST_INVALID_SENTIMENT,
+        TEST_DONE_STRING,
+        TEST_ENTRY_PRICE_STRING,
+        TEST_EXIT_PRICE_STRING,
+        FUTURE_INVALID_DATE_STRING,
+        TEST_REMARK
+    };
+
+    private InstrumentManager instrumentManager;
+    private Storage storage;
+
+    @BeforeEach
+    void initialiseTestResources() {
+        storage = new Storage();
+        instrumentManager = InstrumentManager.getInstance();
+        instrumentManager.addInstrument(TEST_FOREX);
+    }
+
+    @Test
+    void checkNoOfParameters() {
+        String textFileFormattedInstrument = instrumentManager.getInstrument(0).textFileFormatting();
+        String combinedString = TEST_NAME + TEST_SENTIMENT + TEST_REMARK;
+        int lengthOfFormattedInstrument = combinedString.length() + NUMBER_OF_PARAMS + LENGTH_OF_DATE
+                + LENGTH_OF_PRICES + LENGTH_OF_DONE + LENGTH_OF_INSTRUMENT;
+        assertEquals(lengthOfFormattedInstrument, textFileFormattedInstrument.length());
+    }
```

Comment:
should this be in InstrumentEncoderTest? Cause are converting from instrument to save file format.
^ in interest of time I don't think this is very impt